### ISSUE TITLE
Using tf.contrib.training.batch_sequences_with_states with tf.data.Dataset

### ISSUE THREAD

**Issue by npuichigo** [tensorflow/tensorflow#20964]
As far as I know, tf.contrib.training.SequenceQueuingStateSaver relies on queues and queue-runners, so we'll still need to call tf.train.start_queue_runners() when used with the new tf.data.Dataset API.

However, when I use tf.data.Iterator.from_structure to construct the input that is reusable with many different datasets, error occurs unless I initialize the Iterator before calling tf.train.start_queue_runners(). So I need to switch to a certain dataset first, and it's a quite strange usage.

```python
iterator = Iterator.from_structure(tf.int64, tf.TensorShape([]))

dataset_range = Dataset.range(10)
range_initializer = iterator.make_initializer(dataset_range)

dataset_evens = dataset_range.filter(lambda x: x % 2 == 0)
evens_initializer = iterator.make_initializer(dataset_evens)

# Define a model based on the iterator; in this example, the model_fn
# is expected to take scalar tf.int64 Tensors as input (see
# the definition of 'iterator' above).
prediction, loss = model_fn(iterator.get_next())

# I need to pick a certain dataset, and run the initializer first.
sess.run(range_initializer)
tf.train.start_queue_runners(sess=sess, coord=coord)

# Train for `num_epochs`, where for each epoch, we first iterate over
# dataset_range, and then iterate over dataset_evens.
for _ in range(num_epochs):
  # Initialize the iterator to `dataset_range`
  sess.run(range_initializer)
  while True:
    try:
      pred, loss_val = sess.run([prediction, loss])
    except tf.errors.OutOfRangeError:
      break

  # Initialize the iterator to `dataset_evens`
  sess.run(evens_initializer)
  while True:
    try:
      pred, loss_val = sess.run([prediction, loss])
    except tf.errors.OutOfRangeError:
      break
```

By doing so, it seems that the queue will be closed when I switch to another dataset next time.

So will tensorflow add batch_sequences_with_states like support to tf.data.Dataset without relying on the old queue mechanisms and calling tf.train.start_queue_runners explicitly.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by robieta** [comment]
My understanding is that datasets are supposed to replace queue runners, not work in conjunction with them. Why are you not using `dataset.make_one_shot_iterator().get_next()` to get an iterator tensor?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by hansika** [comment]
I have the same question. Since the tf.data.Dataset API is expected to replace the queue runners, is there a functionality similar to batch_sequences_with_states in the Dataset API to support truncated back propagation? Because, clearly the batch_sequences_with_states method needs queue runners to work.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 16 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 31 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 46 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 61 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 76 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 91 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by mohantym** [comment]
We see that you are using old version of Tensorflow which is officially considered as end of life, We recommend that you upgrade to 2.4 or later version and let us know if the issue still persists in newer versions .Please open a new issue in case you face any errors, we will get you the right help .Thanks!
